oxrdflib/parser.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Optional
+
+from rdflib import ConjunctiveGraph, Graph
+from rdflib.exceptions import ParserError
+from rdflib.parser import FileInputSource, InputSource, Parser
+
+from oxrdflib._type import get_mime_type
+from oxrdflib.store import OxigraphStore
+from oxrdflib.utils.converter import to_ox
+
+__all__ = [
+    "OxTurtleParser",
+    "OxNTriplesParser",
+    "OxRdfXmlParser",
+]
+
+
+class OxParser(Parser):
+    def __init__(self):
+        pass
+
+    def parse(
+        self,
+        source: InputSource,
+        sink: Graph | ConjunctiveGraph,
+        format: str,
+        encoding: Optional[str] = "utf-8",
+        **kwargs: Any,
+    ) -> None:
+        if encoding not in [None, "utf-8"]:
+            raise ParserError("N3/Turtle files are always utf-8 encoded, I was passed: %s" % encoding)
+
+        if type(source) not in [FileInputSource, InputSource]:
+            raise ParserError("Source must be either io(bytes) or io(str) or str or pathlib.Path")
+
+        if not isinstance(sink.store, OxigraphStore):
+            raise ParserError(
+                "Sink store must be instance of OxigraphStore, " f"got {sink.store.__class__.__name__} store instead"
+            )
+
+        base_iri = sink.absolutize(source.getPublicId() or source.getSystemId() or "")
+
+        if kwargs.get("transactional", False):
+            sink.store._inner.load(
+                source.file,
+                format,
+                base_iri=base_iri,
+                to_graph=to_ox(sink.identifier),
+            )
+        else:
+            sink.store._inner.bulk_load(
+                source.file,
+                format,
+                base_iri=base_iri,
+                to_graph=to_ox(sink.identifier),
+            )
+
+
+class OxTurtleParser(OxParser):
+    def parse(
+        self,
+        source: InputSource,
+        sink: Graph,
+        format: str = get_mime_type("oxTurtle"),
```

Comment:
Having the format eg `application/n-triples` written explictely would be way more readable imho.

Reply:
Agree, we only need "ox" prefix/namings when we are registering entry-point  to differentiate from rdflib native parsers.

Reply:
updated and simplified